### PR TITLE
Add some DelayedJob instrumentation

### PR DESCRIPTION
**Pull request by hennevogel**
So we can know what's going on in the queue and how jobs perform.

### CODE REVIEW

File: src/api/config/initializers/influxdb_dj_subscriber.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+if CONFIG['influxdb_hosts']
+  ActiveSupport::Notifications.subscribe(/active_job/) do |*args|
+    event = ActiveSupport::Notifications::Event.new(*args)
+    job = event.payload[:job]
+    exception = event.payload[:exception_object]
+
+    def write_queue_stats(state, job)
+      puts "write_queue_stats #{state}"
```
Comment (coolo) [inline src/api/config/initializers/influxdb_dj_subscriber.rb]: why puts and not Rails.logger?

File: src/api/config/initializers/influxdb_dj_subscriber.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      InfluxDB::Rails.client.write_point('active_job.queue',
+                                         tags: { state: state, job: job.class.name, queue: job.queue_name },
+                                         values: { value: 1 })
+    end
+
+    def write_performance_stats(job, event)
+      puts "write_performance_stats #{event.duration}"
```
Comment (coolo) [inline src/api/config/initializers/influxdb_dj_subscriber.rb]: same here

File: src/api/config/initializers/influxdb_dj_subscriber.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      InfluxDB::Rails.client.write_point('active_job.performance',
+                                         tags: { job: job.class.name, queue: job.queue_name },
+                                         values: { value: event.duration })
+    end
+
+    if event.name == 'perform.active_job' && exception
+      InfluxDB::Rails.client.write_point('active_job.queue',
+                                         tags: { state: :failed, job: job.class.name, queue: job.queue_name, id: job.job_id },
```
Comment (coolo) [inline src/api/config/initializers/influxdb_dj_subscriber.rb]: careful! if you use the job id as a tag you're breaking all retention policies as you can no longer group data points.

File: src/api/config/initializers/influxdb_dj_subscriber.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                                         values: { value: 1 })
+      return true
+    end
+
+    case event.name
+    when 'enqueue.active_job'
+      write_queue_stats(:queued, job)
+    # we do not use this currently, we use clockwork instead...
```
Comment (coolo) [inline src/api/config/initializers/influxdb_dj_subscriber.rb]: I don't think this comment is still true

Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: Why?

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: unintentional

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 3 above)
Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: Retention policies are unbreakable :clown_face: This would generate non-downsampleable (cool I just invented a word!)  data, yes.

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 4 above)
Comment (coolo) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: ok, let me rephrase: the comment is very confusing. I bet there are a lot of other things not used you can put here as comments - just to confuse reviewers :)

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 3 above)
Comment (coolo) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: Just put it as additional value

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 4 above)
Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: better?

File: src/api/config/initializers/influxdb_dj_subscriber.rb (hunk 3 above)
Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: yeah ok...

File: src/api/config/initializers/influxdb_dj_subscriber.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      InfluxDB::Rails.client.write_point('active_job.queue',
+                                         tags: { state: :failed, job: job.class.name, queue: job.queue_name },
+                                         values: { value: 1, id: job.job_id })
+      return true
+    end
+
+    case event.name
+    when 'enqueue.active_job'
+      write_queue_stats(:queued, job)
+    # when 'enqueue_at.active_job'
```
Comment (bgeuken) [inline src/api/config/initializers/influxdb_dj_subscriber.rb]: This looks like a leftover

Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: I had a comment explaining why this is here but coolo didn't like it :clown_face:

Comment (coolo) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: But in the end @bgeuken and me have the same problem: it's rather puzzling to mention things not used.

Comment (hennevogel) [reply src/api/config/initializers/influxdb_dj_subscriber.rb]: removed
